extensions/amp-ad/0.1/amp-ad-cross-domain-iframe-handler.js

```diff
@@ -151,28 +158,40 @@ export class AmpAdApiHandler {
     });
   }
 
-  /** @override  */
-  unlayoutCallback() {
+  /**
+   * Return the cross domain iframe
+   * @return {?Element}
+   */
+  getIframe() {
+    return this.iframe_;
+  }
+
+  /**
+   * Cleans up the listeners on the cross domain ad iframe.
+   * And free the iframe resource
+   * @param {boolean=} opt_force
+   */
+  freeIframe(opt_force) {
     this.cleanup_();
     if (this.iframe_) {
-      removeElement(this.iframe_);
-      this.iframe_ = null;
+      if (opt_force || this.iframe_.name.indexOf('_master') == -1) {
```

Comment:
i'll break this line. 
would the check of `_master` exist for A4A friendly iframe case? If so I think it's fine to move this to `amp-ad-ui`. Otherwise I feel this is only related to releasing iframe resource, and I would like to keep the logic in iframe-handler.